REPL.cpp

```diff
@@ -97,37 +109,41 @@ bool REPL::IsExitString(const std::string &line)
     return line == "e" || line == "exit";
 }
 
+//NOTE(sasha): We don't use the normal logging system here because the
+//             DiagnosticEngine will have shown the error.
+#define CHECK_ERROR() if(m_diagnostic_engine.hadAnyError()) { return true; }
+#define PRINT_INVALID_REDECLARATION(name) { std::cout << "Invalid redeclaration of " << name << "\n"; return true; }
+
 bool REPL::ExecuteSwift(std::string line)
 {
-    //NOTE(sasha): We don't use the normal logging system here because the
-    //             DiagnosticEngine will show have showed the error.
-#define CHECK_ERROR() if(m_diagnostic_engine.hadAnyError()) { return true; }
     m_diagnostic_engine.resetHadAnyError();
 
     if(IsExitString(line))
         return false;
 
+    swift::Mangle::ASTMangler mangler;
+
     ReplInput input = AddToSrcMgr(line);
-    auto module_id = m_ast_ctx->getIdentifier(input.module_name);
-    auto *module = swift::ModuleDecl::create(module_id, *m_ast_ctx);
+    auto repl_module_id = m_ast_ctx->getIdentifier("__REPL__");
+    auto *repl_module = swift::ModuleDecl::create(repl_module_id, *m_ast_ctx);
     CHECK_ERROR();
     constexpr auto implicit_import_kind =
         swift::SourceFile::ImplicitModuleImportKind::Stdlib;
     m_invocation.getFrontendOptions().ModuleName = input.module_name.c_str();
     m_invocation.getIRGenOptions().ModuleName = input.module_name.c_str();
 
-    swift::SourceFile *src_file = new (*m_ast_ctx) swift::SourceFile(
-        *module, swift::SourceFileKind::Main, input.buffer_id,
+    swift::SourceFile *tmp_src_file = new (*m_ast_ctx) swift::SourceFile(
+        *repl_module, swift::SourceFileKind::Main, input.buffer_id,
         implicit_import_kind);
```

Comment:
Probably should have some error checking to ensure that `new` didn't fail (and return `nullptr`)